compiler/grainformat/comment_utils.re

```diff
@@ -0,0 +1,564 @@
+          && cmtechar <= stmt_end_char) {
+        [cmt, ...get_comments_inside_location(location, remaining_comments)];
+      } else {
+        get_comments_inside_location(location, remaining_comments);
+      };
+    } else {
+      get_comments_inside_location(location, remaining_comments);
+    };
+  };
+};
+
+let get_comments_between_locations =
+    (~loc1: Location.t, ~loc2: Location.t, comments: list(Parsetree.comment)) => {
+  let (_, stmt_end_line, stmt_end_char, _) =
+    Locations.get_raw_pos_info(loc1.loc_end);
+  let (_, stmt_start_line, stm_start_char, _) =
+    Locations.get_raw_pos_info(loc2.loc_start);
+
+  // invert the request to look inside the location in the gap
+
+  let start_loc: Lexing.position = {
+    pos_fname: "",
+    pos_lnum: stmt_end_line,
+    pos_bol: 0,
+    pos_cnum: stmt_end_char,
+  };
+  let end_loc: Lexing.position = {
+    pos_fname: "",
+    pos_lnum: stmt_start_line,
+    pos_bol: 0,
+    pos_cnum: stm_start_char,
+  };
+
+  let location: Location.t = {
+    loc_start: start_loc,
+    loc_end: end_loc,
+    loc_ghost: false,
+  };
+  get_comments_inside_location(~location, comments);
+};
+
+let get_comments_enclosed_and_before_location =
+    (~loc1: Location.t, ~loc2: Location.t, comments: list(Parsetree.comment)) => {
+  let (_, loc_start_line, loc_start_char, _) =
+    Locations.get_raw_pos_info(loc1.loc_start);
+  let (_, stmt_start_line, stm_start_char, _) =
+    Locations.get_raw_pos_info(loc2.loc_start);
+
+  // invert the request to look inside the location in the gap
+
+  let start_loc: Lexing.position = {
+    pos_fname: "",
+    pos_lnum: loc_start_line,
+    pos_bol: 0,
+    pos_cnum: loc_start_char,
+  };
+  let end_loc: Lexing.position = {
+    pos_fname: "",
+    pos_lnum: stmt_start_line,
+    pos_bol: 0,
+    pos_cnum: stm_start_char,
+  };
+
+  let location: Location.t = {
+    loc_start: start_loc,
+    loc_end: end_loc,
+    loc_ghost: false,
+  };
+
+  get_comments_inside_location(~location, comments);
+};
+
+let get_comments_from_start_of_enclosing_location =
+    (
+      ~wrapper: Location.t,
+      ~location: Location.t,
+      comments: list(Parsetree.comment),
+    ) => {
+  let (_, wrap_start_line, wrap_start_char, _) =
+    Locations.get_raw_pos_info(wrapper.loc_start);
+  let (_, loc_start_line, loc_start_char, _) =
+    Locations.get_raw_pos_info(location.loc_start);
+
+  let start_loc: Lexing.position = {
+    pos_fname: "",
+    pos_lnum: wrap_start_line,
+    pos_bol: 0,
+    pos_cnum: wrap_start_char,
+  };
+  let end_loc: Lexing.position = {
+    pos_fname: "",
+    pos_lnum: loc_start_line,
+    pos_bol: 0,
+    pos_cnum: loc_start_char,
+  };
+
+  let location: Location.t = {
+    loc_start: start_loc,
+    loc_end: end_loc,
+    loc_ghost: false,
+  };
+  get_comments_inside_location(~location, comments);
+};
+
+let get_comments_between_locs =
+    (~loc1: Location.t, ~loc2: Location.t, comments: list(Parsetree.comment)) => {
```

Comment:
And in that vein, should these be `begin_loc` and `end_loc`?

Reply:
Done.